Devel branch work started

AmritPatel opened the pull request:
1. Prettified source using https://black.now.sh/
2. Removed 3 second wait between Zillow server requests
3. Removed old commented code
4. Removed web browser open as serves no purpose
5. Added output dump to file

AmritPatel committed:
Devel branch work started

1. Prettified source using https://black.now.sh/
2. Removed 3 second wait between Zillow server requests
3. Removed old commented code
4. Removed web browser open as serves no purpose
5. Added output dump to file

MattThomas9 commented on res.inp:
```diff
@@ -7,4 +7,4 @@ State:              Maryland
 [Search Control]
-Search Box Half-Width (miles): 0.5
\ No newline at end of file
+Search Box Half-Width (miles): 0.1
```
Ok good job @AmritPatel, we'll keep this at 0.1 during dev stages so it runs quicker.

MattThomas9 commented on rescrape.py:
```diff
@@ -1,31 +1,41 @@
 import webbrowser
 import requests
 from bs4 import BeautifulSoup
-import time
```
I tried running the code on 0.5 search radius (lots of http requests) with the 3 second sleep between each request commented out and it ran fine. I can't remember exactly what was causing my issues originally that made me put it in there. So we can take this out for right now.

MattThomas9 commented on rescrape.py:
```diff
@@ -36,92 +46,117 @@
-             '"isListVisible":true}']
-
-# Open up the Zillow web page to view the search box, results, and see how many pages there are.
-webbrowser.open_new(ZillowURL[0])
```
I want this to be left in the code. I like how it opens up the main Zillow search page so the user can visualize what they are searching on. But more importantly, when Zillow changes their HTML, and I need to get to the website quickly to determine what the new HTML is for scraping purposes, having the exact web page already open allows me to quickly view the updated HTML that I will end up using in the code.

MattThomas9 replied:
As a matter of fact, disregard. Don't need to open the web page. Can simply just print it to screen so if the user wants to view it, they can click the link that comes up (or c&P it into their own browser).

MattThomas9 replied:
Disregard again. Printing URL to screen only hyperlinks part of it because there are "{" characters in it. So pycharm doesn't recognize the parts behind the "{" as also part of the hyperlink. Which means it makes it 1.) look like shit, and 2.) unclickable. Therefore, will put webbrowser.open back in. I think its alright. Just minimize it if you don't want to see it. sheesh.

AmritPatel replied:
> Disregard again. Printing URL to screen only hyperlinks part of it because there are "{" characters in it. So pycharm doesn't recognize the parts behind the "{" as also part of the hyperlink. Which means it makes it 1.) look like shit, and 2.) unclickable. Therefore, will put webbrowser.open back in. I think its alright. Just minimize it if you don't want to see it. sheesh.

From  [Python documentation for format string syntax](https://docs.python.org/library/string.html#formatstrings):

> Format strings contain “replacement fields” surrounded by curly braces `{}`. Anything that is not contained in braces is considered literal text, which is copied unchanged to the output. If you need to include a brace character in the literal text, it can be escaped by doubling: `{{` and `}}`.

AmritPatel replied:
> I want this to be left in the code. I like how it opens up the main Zillow search page so the user can visualize what they are searching on. But more importantly, when Zillow changes their HTML, and I need to get to the website quickly to determine what the new HTML is for scraping purposes, having the exact web page already open allows me to quickly view the updated HTML that I will end up using in the code.

I think this is better left as a debugging/development tool...

MattThomas9 commented on rescrape.py:
```diff
@@ -203,38 +237,51 @@
 
 for i in range(0, len(SoldHomeZillowLinks)):
 
-    # A delay of 3 seconds between each request is instituted to ensure the Zillow servers are not overwhelmed.
```
As mentioned above, we can get rid of this right now. But note that if we start seeing HTTP request issues, this may be a good place to start for troubleshooting.

MattThomas9 reviewed (approved):
Well done, @AmritPatel

MattThomas9 commented:
@AmritPatel, I know you said above that you "5. Added output dump to file", but I didn't review any change in this commit that dumped output to file...

MattThomas9 commented:
@AmritPatel, also, is it your intention to keep this devel branch going? Because Github is telling me that now with the pull request successfully merged and closed, it "can now be safely deleted." What to do?

AmritPatel commented:
> @AmritPatel, I know you said above that you "5. Added output dump to file", but I didn't review any change in this commit that dumped output to file...

Yea, I added by mistake as I didn't get that far.

AmritPatel commented:
> @AmritPatel, also, is it your intention to keep this devel branch going? Because Github is telling me that now with the pull request successfully merged and closed, it "can now be safely deleted." What to do?

I think we should establish the practice of deleting all merged branches to keep things nice and tidy.
